### ISSUE TITLE
Quicksight allowed_to_* fails when username has "/" (blocking)

### ISSUE THREAD

**Issue by brombach** [awslabs/aws-data-wrangler#434]
**Describe the bug**
When a quicksight username includes a "/" character, the following error is incorrectly thrown.
```
botocore.errorfactory.InvalidParameterValueException: An error occurred (InvalidParameterValueException) when calling the CreateDataSet operation: One or more principals in resource permissions list are not valid QuickSight users
```

**To Reproduce**
The output of `wr.quicksight.list_users` includes an entry structured like this: `{'username': 'example-role/exampleuser', 'role': 'ROLE'}`. Using exactly that username, _example-role/exampleuser_, the error above is thrown. No subset of the username, such as exampleuser, works, nor does using the full ARN (arn:aws:quicksight:region:111111111111:user/default/example-role/exampleuser). 

**Likely Issue**
Following this stack trace:
```
File "datawrangler.py", line 69, in <module>
    allowed_to_manage=QUICKSIGHT_ACCOUNT_NAMES,
  File "<>/python3.7/site-packages/awswrangler/quicksight/_create.py", line 344, in create_athena_dataset
    client.create_data_set(**args)
  File "<>/python3.7/site-packages/botocore/client.py", line 337, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "<>/python3.7/site-packages/botocore/client.py", line 656, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.InvalidParameterValueException: An error occurred (InvalidParameterValueException) when calling the CreateDataSet operation: One or more principals in resource permissions list are not valid QuickSight users
````

reveals that the error is in the way that ARNs are constructed for quicksight principals in `_generate_principal` [here](https://github.com/awslabs/aws-data-wrangler/blob/b580326b5a5998b8ee375b7ca36b52542a26f2c0/awswrangler/quicksight/_create.py#L55)
```
def _generate_principal(user_name: str, account_id: str, region: str) -> str:
    user_name = user_name if "/" in user_name else f"default/{user_name}"
    return f"arn:aws:quicksight:{region}:{account_id}:user/{user_name}"
```
which does not add "default/" if there is already a slash in the username. Adding "default/" to the username manually prevents the retrieval of account_id and region and fails identically.

**Comment by igorborgest** [comment]
Thank you @brombach. Really useful troubleshooting!

I've just updated the approach and now you can simply pass the usernames and Wrangler will fetch the related ARNs from the QuickSight API. (See the commit above for more details)

Do you mind to install it directly from our repository and check if it is everything fine for your use case?
The idea is to publish it in the version `1.10.0` this weekend.

```sh
pip install git+https://github.com/awslabs/aws-data-wrangler.git --no-use-pep517
```

**Comment by brombach** [comment]
Worked like a charm! Thanks for the quick fix.

**Comment by igorborgest** [comment]
Released on [1.10.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.10.0)
